gymnasium/wrappers/vector/vectorize_observation.py

```diff
@@ -53,10 +86,49 @@ def single_observation(self, observation: ObsType) -> ObsType:
 
 
 class VectorizeTransformObservation(VectorObservationWrapper):
-    """Vectorizes a single-agent transform observation wrapper for vector environments."""
+    """Vectorizes a single-agent transform observation wrapper for vector environments.
+
+    Most of the lambda observation wrappers for single agent environments have vectorized implementations,
+    it is advised that users simply use those instead via importing from `gymnasium.wrappers.vector...`.
+    The following example illustrate use-cases where a custom lambda observation wrapper is required.
+
+    Example:
+        The normal observation:
+        >>> import gymnasium as gym
+        >>> envs = gym.make_vec("CartPole-v1", num_envs=3)
+        >>> obs, info = envs.reset(seed=123)
+        >>> envs.close()
+        >>> obs
+        array([[ 0.01823519, -0.0446179 , -0.02796401, -0.03156282],
+               [ 0.02852531,  0.02858594,  0.0469136 ,  0.02480598],
+               [ 0.03517495, -0.000635  , -0.01098382, -0.03203924]],
+              dtype=float32)
+
+        Applying a custom lambda observation wrapper that duplicates the observation from the environment
+        >>> import numpy as np
+        >>> import gymnasium as gym
+        >>> from gymnasium.spaces import Box
+        >>> from gymnasium.wrappers import TransformObservation
+        >>> envs = gym.make_vec("CartPole-v1", num_envs=3)
+        >>> old_space = envs.single_observation_space
+        >>> new_space = Box(low=np.array([old_space.low, old_space.low]), high=np.array([old_space.high, old_space.high]))
+        >>> envs = VectorizeTransformObservation(envs, wrapper=TransformObservation, func=lambda x: np.array([x, x]), observation_space=new_space)
+        >>> obs, info = envs.reset(seed=123)
+        >>> envs.close()
+        >>> obs
+        array([[[ 0.01823519, -0.0446179 , -0.02796401, -0.03156282],
+                [ 0.01823519, -0.0446179 , -0.02796401, -0.03156282]],
+        <BLANKLINE>
+               [[ 0.02852531,  0.02858594,  0.0469136 ,  0.02480598],
+                [ 0.02852531,  0.02858594,  0.0469136 ,  0.02480598]],
+        <BLANKLINE>
+               [[ 0.03517495, -0.000635  , -0.01098382, -0.03203924],
+                [ 0.03517495, -0.000635  , -0.01098382, -0.03203924]]],
+              dtype=float32)
+    """
 
-    class VectorizedEnv(Env):
-        """Fake single-agent environment uses for the single-agent wrapper."""
+    class _SingleEnv(Env):
```

Comment:
Is there a reason why you changed the name? I think the name is better but is there any other reason?

Reply:
Fair, that makes sense